src/ws.nim

```diff
@@ -254,7 +259,7 @@ proc send*(ws: WebSocket, text: string, opcode = Opcode.Text): Future[void] {.as
       let data = frame[i ..< min(frame.len, i + maxSize)]
       discard await ws.tcpSocket.write(data)
       i += maxSize
-      await sleepAsync(1)
+      await stepsAsync(1)
```

Comment:
there should be no steps or awaits here - `await write` already does all the necessary waiting

Reply:
Yes, we've already pointed this out in #1. There might be some exceptions for this rule, when for example there is a chance of ending up in a tight synchronous loop that can starve the main event loop, as a sort of a `yield`, but those are rare and for the most part there is never a need for such a `yield`.

Reply:
this still needs to be resolved.

Reply:
ie, the `sleepAsync` needs to go